### PR TITLE
[6106,6284] Introduce phony targets for building certain sets of targets and update non-package-install target dependencies

### PR DESCRIPTION
**Pull request by SwooshyCueb**
In service of 6106 and 6284

This PR does three things:
1. Corrects an indentation size inconsistency left over from when I rebased #6201 prior to merging.
2. Creates a few phony targets that depend on different sets of targets.
3. Updates the list of dependency targets for non-package-install targets using the new phony targets.

### CODE REVIEW

**Comment by SwooshyCueb** [comment]
A little extra background for this PR:

I noticed we were using target names for the `DEPENDS` arguments in the `add_custom_target` calls creating the non-package-install targets.
Per the [CMake docuemntation of `add_custom_target`](https://cmake.org/cmake/help/v3.12/command/add_custom_target.html), the `DEPENDS` argument(s) should not reference other *targets*:
> `DEPENDS`
> &emsp;Reference files and outputs of custom commands created with [`add_custom_command()`](https://cmake.org/cmake/help/v3.12/command/add_custom_command.html#command:add_custom_command) command calls in the same directory (`CMakeLists.txt` file).
> &emsp;...
> &emsp;Use the [`add_dependencies()`](https://cmake.org/cmake/help/v3.12/command/add_dependencies.html#command:add_dependencies) command to add dependencies on other targets.

The fact that passing target names as `DEPENDS` arguments even works at all is due to an implementation detail of the CMake generators we use.

Initially, my intention was simply add in the call to `add_dependencies` and replace the long list of targets with `all`. Quick and easy.
I would not be so lucky. `add_dependencies` does not work with CMake-generated targets. We still have to pass in targets we declare.
But the big list of dependency targets is still a maintenance hazard, so I created the phony targets to "accumulate" installed targets.

**Review by korydraughn — COMMENTED** [review]
Everything looks good.

Are there any remaining TODOs for this PR?

File: CMakeLists.txt [hunk 1]
```diff
@@ -306,50 +306,22 @@ foreach(DATABASE_PLUGIN postgres oracle mysql)
...
-    irods_auth_legacy_plugin-pam_client
-    irods_auth_legacy_plugin-pam_server
-    irods_network_plugin-ssl_client
-    irods_network_plugin-ssl_server
-    irods_network_plugin-tcp_client
-    irods_network_plugin-tcp_server
+  )
+  add_dependencies(
+    non-package-install-${DATABASE_PLUGIN}
+    all-lib
```
[the remark this answers is not in the document]
Comment (SwooshyCueb) [reply CMakeLists.txt:312]: I'm only in favor of prefixing our targets in this way for targets that meet a couple of criteria:
- It is unlikely that a developer would wish to call the target directly
- The target has a generic name that is likely to also be used by superprojects or siblings within a superproject

**Comment by SwooshyCueb** [comment]
> Are there any remaining TODOs for this PR?

Only the change you requested

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (alanking) [reply CMakeLists.txt:312]: That makes sense to me. Thanks!

**Review by korydraughn — APPROVED** [review]

**Review by alanking — APPROVED** [review]
Good stuff. I think we're good to go

**Comment by SwooshyCueb** [comment]
\#'d

**Pull request merged**